### ISSUE TITLE
Soft limit for maxEventCount

### ISSUE THREAD

**Issue by noxhirsch** [marksie1988/atomic-calendar-revive#311]
## Summary
When you have e.g. 4 events per day, set `maxDaysToShow: 2` and `maxEventCount: 6` you'll have only half of tomorrows events listed. If you could set `maxEventSoftCount: 6`, tomorrow would either not shown or shown completely (depending on the way the function would be implemented)

## Motivation
This is especially relevant if you have multiple calendars in the card or lots of events per day in one calendar. With the current hard limit, it can give the impression, that you only have e.g. 2 events for the last day shown in the calendar card while instead you have e.g. 4 events. If you look on the calendar card to plan something for the last day shown, you could miss an already planned event.

## Describe alternatives you've considered
Instead of another limit setting, there could be a soft-limit-flag:
```
maxEventCount: 6
maxEventSoft: true
```

**Comment by marksie1988** [comment]
Hi,

 I don't understand why you would want to have a limit of 6 events but then see all the events for a day? If you want to see all the events for the two days you wouldn't set the maxEventCount 

Or am I missunderstanding why you would want this feature?

**Comment by noxhirsch** [comment]
Hi,
maybe my example was bad - I thought it would be more relevant to people with many events per day.
I'll try to explain how I'd use it (and was using with "ljmerza/calendar-card"):
I have a calendar card on my dashboard which listed around 10 calendar entries all the time. Sometimes I have e.g. 3 events a day and other times I have only two events in a week. If I only set e.g. `maxDaysToShow: 7` that calendar card would sometimes be very long and you'd have to scroll far down to see the other cards - or sometimes half of the dashboard would be empty.
On the other hand if I set `maxEventCount: 10`, I'd never know if all events of the last day are shown in the calendar card.
If there's a soft limit, the length of the calendar card wouldn't vary that much, but there wouldn't be that "useless" incomplete last day in the card. So it's kind of a compromise between those settings.

**Comment by marksie1988** [comment]
I've tried to understand this however I'm still not sure what you mean. 

If I have say `maxDaysToShow: 7` and `maxEventCount: 10` with `maxEventSoft: true` and say the soft limit shows all the events for the day, you would still have a long list as it would be the equivalent of not setting `maxEventCount`

I think a better solution would be that if `maxEventCount: 10` is set then a new text appears at the bottom saying "x events are not listed" then on clicking it opens the calendar, or something similar.

**Comment by noxhirsch** [comment]
> If I have say `maxDaysToShow: 7` and `maxEventCount: 10` with `maxEventSoft: true` and say the soft limit shows all the events for the day, you would still have a long list as it would be the equivalent of not setting `maxEventCount`

Example A) If you have many events on one day, the list would indeed get very long:
| maxEventCount: 5<br>SoftLimit: off | maxEventCount: 10<br>SoftLimit: on                                     |
|------------------------------------|------------------------------------------------------------------------|
| Event 1<br>Event 2               | Event 1<br>Event 2                                                   |
| Event 3<br>Event 4               | Event 3<br>Event 4                                                   |
| Event 5                           | Event 5<br>Event 6<br>Event 7<br>Event 8<br>Event 9 <br>Event 10 <br>Event 11 <br>Event 12 <br>Event 13 <br>Event 14 <br>Event 15 |

Example B) If you have only a few events per day, it doesn't make a big difference:
| maxEventCount: 5<br>SoftLimit: off | maxEventCount: 10<br>SoftLimit: on                                     |
|------------------------------------|------------------------------------------------------------------------|
| Event 1<br>Event 2               | Event 1<br>Event 2                                                   |
| Event 3<br>Event 4               | Event 3<br>Event 4                                                   |
| Event 5                           | Event 5<br>Event 6 |

> I think a better solution would be that if `maxEventCount: 10` is set then a new text appears at the bottom saying "x events are not listed" then on clicking it opens the calendar, or something similar.

I think for Example A, your idea would be the perfect solution - the list doesn't get longer but you now see that X events are missing
If like in Example B only one or two events are missing it would be better to show the events directly, in my opinion (so you would not have to click a link to only show one missing event).

To sum it up: For me, the perfect solution would be if a soft limit option exists and the "missing events counter" would be shown if softLimit is disabled - but of course I understand that your time is limited and not every feature can be implemented just because some random dude on the internet wants it ;) (I used the now archived ljmerza/calendar-card before which had that option and it's always quite an adjustment if you got used to something) I already tried to do it myself, so I can open a pull request myself, but sadly I'm to dumb to understand the code...

TL;DR: Your idea would solve the problem I explained in "Motivation" of this issue, so it is indeed a good solution 👍

**Comment by marksie1988** [comment]
I also find JS a headache, I’m mainly a backend developer so it stumps me quite a bit! 

Ok I think I understand now and also took a quick look at the other card you mentioned. 

So I think I will add two new options:
`softLimit: 3` This would then show 3 events over the limit if 3 or less exist, in a case where more exist a link will be shown. 

Where no softLimit is set then it will just show a link and count of missing events. 

I will see what I can do :)

**Comment by noxhirsch** [comment]
A non-binary `softLimit` is an great idea  👍👍👍 Thank you very much! :)

**Comment by marksie1988** [comment]
In testing I have the following: 

```
maxDaysToShow: 7
maxEventCount: 3
```
![image](https://user-images.githubusercontent.com/159166/103529238-3874f680-4e7d-11eb-8c00-58ab49b712f4.png)


```
maxDaysToShow: 7
maxEventCount: 3
softLimit: 1
```
![image](https://user-images.githubusercontent.com/159166/103528860-98b76880-4e7c-11eb-83ad-91411577cc6a.png)

```
maxDaysToShow: 7
maxEventCount: 6
softLimit: 1
```
![image](https://user-images.githubusercontent.com/159166/103528937-bab0eb00-4e7c-11eb-87b8-bbc41ad909a0.png)

```
maxDaysToShow: 7
maxEventCount: 6
softLimit: 3
```
![image](https://user-images.githubusercontent.com/159166/103529008-d4523280-4e7c-11eb-9537-2161d861cab5.png)

**Comment by noxhirsch** [comment]
That's amazing! Thank you very much! :)
